Adding Trainer

captain-pool opened the pull request:
Adding Trainer for Knowledge Distillation.

Addresses #42, #45 , #46 , #47, #44

srjoglekar246 commented on E3_Distill_ESRGAN/libs/models/teacher.py:
```diff
@@ -1,5 +1,12 @@
 import os
 import sys
+from libs import settings
 # Fetching Generator from ESRGAN
```
newline between imports and code.

srjoglekar246 commented on E3_Distill_ESRGAN/libs/settings.py:
```diff
@@ -1,3 +1,4 @@
+""" Module Containing Settings and Training Statistics Model YAML Handler """
```
More readable docstring please :-)

srjoglekar246 commented on E3_Distill_ESRGAN/libs/train.py:
```diff
@@ -1 +1,230 @@
+        summary_writer: tf.summary.SummaryWriter object for writing
+                         summary for Tensorboard.
+        data_dir: Location of the stored dataset.
+        manual: Indicate if data_parameter contains Raw Files or TFRecords.
```
nit: Keep 'similar'-types arguments together. So group together `data_dir` and `model_dir`. Also, `manual` doesn't seem the best way to describe what that parameter indicates. Maybe just `raw_data`. And also `data_parameter` -> `data_dir`?

captain-pool replied:
Any suggestions how to indicate if it is `data_dir` or `manual_dir`, without using `manual`?

srjoglekar246 replied:
Wouldn't it be either of the two? So you can check which is not-empty?

captain-pool replied:
okay! Cool. :)

captain-pool replied:
PS: manual_dir and data_dir are parameters of `tfds.load(...)`.

srjoglekar246 commented on E3_Distill_ESRGAN/libs/utils.py:
```diff
@@ -1,5 +1,46 @@
+from lib.utils import RelativisticAverageLoss
+
+
+def save_checkpoint(checkpoint, name, basepath="", student=False):
```
`student` -> `use_student_settings`.

same below.

captain-pool commented:
> Have we added any README or documentation that references the paper we are using for this? And what changes we are making to the loss function, etc?

We have 2 options for Loss. We can choose any of these, depending on the output we are getting.
- Comparative Loss -MSE Loss (As Stated by the Paper).
- **[Preferred]** Adversarial Loss - Joint MSE and Relativistic Average Loss.

I've implemented both, we will be using each of them based on the output we get.
